hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -37,6 +40,58 @@ func machineHealthCheck(nodePool *hyperv1.NodePool, controlPlaneNamespace string
 	}
 }
 
+func KubevirtMachineTemplate(infraName string, hostedCluster *hyperv1.HostedCluster, nodePool *hyperv1.NodePool, controlPlaneNamespace string) (*capikubevirt.KubevirtMachineTemplate, string) {
+	guestQuantity := apiresource.MustParse("4Gi")
+	kubevirtMachineTemplate := &capikubevirt.KubevirtMachineTemplate{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				nodePoolAnnotation: ctrlclient.ObjectKeyFromObject(nodePool).String(),
+			},
+			Namespace: controlPlaneNamespace,
+		},
+		Spec: capikubevirt.KubevirtMachineTemplateSpec{
+			Template: capikubevirt.KubevirtMachineTemplateResource{
+				Spec: capikubevirt.KubevirtMachineSpec{
+					VMSpec: kubevirtv1.VirtualMachineInstanceSpec{
+						Domain: kubevirtv1.DomainSpec{
+							CPU:    &kubevirtv1.CPU{Cores: 2},
+							Memory: &kubevirtv1.Memory{Guest: &guestQuantity},
+							Devices: kubevirtv1.Devices{
+								Disks: []kubevirtv1.Disk{
+									{
+										Name: "containervolume",
+										DiskDevice: kubevirtv1.DiskDevice{
+											Disk: &kubevirtv1.DiskTarget{
+												Bus: "virtio",
+											},
+										},
+									},
+								},
+							},
+						},
+						Volumes: []kubevirtv1.Volume{
+							{
+								Name: "containervolume",
+								VolumeSource: kubevirtv1.VolumeSource{
+									ContainerDisk: &kubevirtv1.ContainerDiskSource{
+										// TODO nargaman - replace with pre-defined image
+										Image: "quay.io/containerdisks/rhcos:4.9",
```

Comment:
With the aws provider, hypershift is attempting to automatically detect the correct AMI based on the release. If we are mirroring/publishing rhcos container disks to a predictable place that we control, maybe we follow the same pattern and search for images in the public quay.io repo.

Reply:
Currently removed the container disk from the code and the user must provide this in CLI
When we will have more official image, will add it as default value